src/CognitiveServices/SpeechRecognition.ts

```diff
@@ -0,0 +1,145 @@
+                    this.log('ListeningStartedEvent');
+                    break;
+                case 'RecognitionStartedEvent':
+                    this.log('RecognitionStartedEvent');
+                    if (this.onAudioStreamingToService) {
+                        this.onAudioStreamingToService()
+                    }
+                    this.isStreamingToService = true;
+                    break;
+                case 'SpeechStartDetectedEvent':
+                    this.log('SpeechStartDetectedEvent');
+                    break;
+                case 'SpeechHypothesisEvent':
+                    let hypothesisEvent = event as CognitiveSpeech.SpeechHypothesisEvent;
+                    this.log('SpeechHypothesisEvent: ' + hypothesisEvent.Result.Text);
+                    if (this.onIntermediateResult) {
+                        this.onIntermediateResult(hypothesisEvent.Result.Text);
+                    }
+                    break;
+                case 'SpeechEndDetectedEvent':
+                    this.log('SpeechEndDetectedEvent');
+                    break;
+                case 'SpeechSimplePhraseEvent':
+                    this.log('SpeechSimplePhraseEvent');
+                    let simplePhraseEvent = event as CognitiveSpeech.SpeechSimplePhraseEvent;
+                    if (CognitiveSpeech.RecognitionStatus[simplePhraseEvent.Result.RecognitionStatus] as any === CognitiveSpeech.RecognitionStatus.Success) {
+                        if (this.onFinalResult) {
+                            this.onFinalResult(simplePhraseEvent.Result.DisplayText);
+                        }
+                    } else {
+                        this.log('Recognition Status: ' + simplePhraseEvent.Result.RecognitionStatus.toString());
+                    }
+                    break;
+                case 'SpeechDetailedPhraseEvent':
+                    this.log('SpeechDetailedPhraseEvent');
+                    break;
+                case 'RecognitionEndedEvent':
+                    this.log('RecognitionEndedEvent');
+                    this.isStreamingToService = false;
+                    break;
+            }
+        }
+
+        let speechContext = null;
+        if (this.referenceGrammarId) {
+            speechContext = JSON.stringify({
+                dgi: {
+                    Groups: [
+                        {
+                            Type: "Generic",
+                            Hints: { ReferenceGrammar: this.referenceGrammarId }
+                        }
+                    ]
+                }
+            });
+        }
+
+        this.actualRecognizer.Recognize(eventhandler, speechContext);
+    }
+
+    public stopRecognizing() {
+        if (this.actualRecognizer != null) {
+            this.actualRecognizer.AudioSource.TurnOff();
+        }
+        this.isStreamingToService = false;
+    }
+
+    private log(message: string) {
+        konsole.log('CognitiveServicesSpeechRecognizer: ' + message);
+    }
+}
```

Comment:
Should probably have newlines at the end of the files. I thought most git clients fixed this up by default?